Cannot use "extends" within package.json

Not sure if this is me being stupid or not - but I cannot seem to load the `extends` directive within the package.json `eslintConfig`.

If I have an `.eslintrc` file with the following:

```json
{
    "extends": "strict"
}
```

All works fine. Now if I try to put that in my package.json:

```json
"eslintConfig": {
  "extends": "strict"
}
"devDependencies": {
  "eslint-config-strict": "^1.1.0"
}
```

It reverts to the default eslint configuration - i.e. not extending from the strict config. I couldn't read anything in the docs suggesting this was not allowed.

I've trimmed out sensitive info. The file is a bit of a mess so produces errors in both default and the `eslint-config-strict` settings, but you can clearly see the package.json is ignoring the extends directive:

```bash
$ cat package.json | grep eslintConfig -B1 -A4
  },
  "eslintConfig": {
    "extends": "strict"
  }
}

$ eslint . # with package.json
  eslint:cli Running on files +0ms
  eslint:cli-engine Processing <file> +6ms
  eslint:cli-engine Linting /Users/<file> +3ms
  eslint:config Constructing config for /Users/<file> +1ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /Users/<folder>/package.json +1ms
  eslint:config Using /Users/<folder>/package.json +4ms
  eslint:config Merging command line environment settings +9ms

<file>
   2:11  error  "require" is not defined                       no-undef
   2:19  error  Strings must use doublequote                   quotes
   3:31  error  "process" is not defined                       no-undef
   4:9   error  "require" is not defined                       no-undef
   4:17  error  Strings must use doublequote                   quotes
   5:18  error  "require" is not defined                       no-undef
   5:26  error  Strings must use doublequote                   quotes
   7:0   error  Missing "use strict" statement                 strict
   9:52  error  object is already declared in the upper scope  no-shadow
  10:39  error  Strings must use doublequote                   quotes
  19:17  error  "require" is not defined                       no-undef
  21:17  error  "file" is not defined                          no-undef
  23:20  error  "require" is not defined                       no-undef
  23:28  error  Strings must use doublequote                   quotes
  26:4   error  Strings must use doublequote                   quotes
  27:4   error  Strings must use doublequote                   quotes
  28:4   error  Strings must use doublequote                   quotes
  29:4   error  Strings must use doublequote                   quotes
  30:4   error  Strings must use doublequote                   quotes
  31:4   error  Strings must use doublequote                   quotes
  32:4   error  Strings must use doublequote                   quotes
  33:4   error  Strings must use doublequote                   quotes
  34:4   error  Strings must use doublequote                   quotes
  35:4   error  Strings must use doublequote                   quotes
  35:13  error  Unexpected trailing comma                      comma-dangle
  37:0   error  "require" is not defined                       no-undef
  37:8   error  Strings must use doublequote                   quotes
  37:19  error  "file" is not defined                          no-undef

✖ 28 problems (28 errors, 0 warnings)
```

Here's with the .eslintrc in place:

```bash
$ cat .eslintrc
{
    "extends": "strict"
}

$ eslint . # with .eslintrc

  eslint:cli Running on files +0ms
  eslint:cli-engine Processing <file> +7ms
  eslint:cli-engine Linting /Users/<file> +2ms
  eslint:config Constructing config for /Users/<file> +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /Users/<folder>/.eslintrc +1ms
  eslint:config Using /Users/<folder>/.eslintrc +8ms
  eslint:config Loading /Users/<folder>/package.json +0ms
  eslint:config Using /Users/<folder>/package.json +2ms
  eslint:config Merging command line environment settings +11ms
  eslint:cli-engine Load plugin filenames +6ms
  eslint:cli-engine Load plugin one-variable-per-var +0ms

<file>
   2:0   error  Unexpected var, use let or const instead           no-var
   3:0   error  Unexpected var, use let or const instead           no-var
   4:0   error  Unexpected var, use let or const instead           no-var
   5:0   error  Unexpected blank line after variable declarations  newline-after-var
   5:0   error  Unexpected var, use let or const instead           no-var
   8:2   error  Expected indentation of 2 characters               indent
   8:4   error  Assignment to function parameter 'object'          no-param-reassign
   9:2   error  Expected indentation of 2 characters               indent
   9:42  error  Missing function expression name                   func-names
   9:52  error  object is already declared in the upper scope      no-shadow
  10:4   error  Expected indentation of 4 characters               indent
  11:6   error  Expected indentation of 6 characters               indent
  12:4   error  Expected indentation of 4 characters               indent
  13:6   error  Expected indentation of 6 characters               indent
  14:4   error  Expected indentation of 4 characters               indent
  15:4   error  Expected indentation of 4 characters               indent
  16:2   error  Expected indentation of 2 characters               indent
  19:0   error  Unexpected blank line after variable declarations  newline-after-var
  19:0   error  Unexpected var, use let or const instead           no-var
  21:17  error  "file" is not defined                              no-undef
  23:0   error  Unexpected var, use let or const instead           no-var
  37:19  error  "file" is not defined                              no-undef

✖ 22 problems (22 errors, 0 warnings)
```

FWIW this is using eslint 0.21.1 but it seems like the issue is still relevant to 22.1.

--- a/lib/config.js
+++ b/lib/config.js
@@ -77,6 +77,10 @@ function loadConfig(filePath) {
                 throw e;
             }
 
+            if (path.basename(filePath) === PACKAGE_CONFIG_FILENAME) {
+                config = config[PACKAGE_CONFIG_FIELD_NAME] || {};
+            }
+
         } else {
 
             // it's a package
@@ -220,13 +224,9 @@ function getLocalConfig(thisConfig, directory) {
         debug("Loading " + localConfigFile);
         localConfig = loadConfig(localConfigFile);
 
-        if (path.basename(localConfigFile) !== LOCAL_CONFIG_FILENAME) {
-
-            // Don't consider a local config file found if the package.json doesn't have the eslintConfig field.
-            if (!localConfig.hasOwnProperty(PACKAGE_CONFIG_FIELD_NAME)) {
-                continue;
-            }
-            localConfig = localConfig[PACKAGE_CONFIG_FIELD_NAME] || {};
+        // Don't consider a local config file found if the config is empty.
+        if (!Object.keys(localConfig).length) {
+            continue;
         }
 
         found = true;

--- a/tests/fixtures/config-extends/package.json
+++ b/tests/fixtures/config-extends/package.json
@@ -1,0 +1,7 @@
+{
+    "name": "test",
+    "version": "0.1.1",
+    "eslintConfig": {
+        "extends": "./subdir/.eslintrc"
+    }
+}

--- a/tests/lib/config.js
+++ b/tests/lib/config.js
@@ -738,6 +738,23 @@ describe("Config", function() {
             assertConfigsEqual(expected, actual);
         });
 
+        it("should extend package configuration from package.json file", function() {
+            var configPath = path.resolve(__dirname, "../fixtures/config-extends/package.json"),
+                configHelper = new Config({ reset: true, useEslintrc: true }),
+                expected = {
+                    rules: { "quotes": [1, "single"], "yoda": 2 },
+                    env: { "browser": true }
+                };
+            // Reason to override this function is force the execution to go into getLocalConfig function and which handles package.json
+            sinon.stub(configHelper, "findLocalConfigFiles").returns([
+                path.resolve(__dirname, "../fixtures/config-extends/package.json")
+            ]);
+
+            var actual = configHelper.getConfig(configPath);
+
+            assertConfigsEqual(expected, actual);
+        });
+
         // Non-recursive extends
         it("should extend recursively defined configuration files", function() {
             var configPath = path.resolve(__dirname, "..", "fixtures", "config-extends", "deep.json"),

